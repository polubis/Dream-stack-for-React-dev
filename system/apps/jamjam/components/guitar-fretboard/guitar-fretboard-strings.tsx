import type { GuitarFretboardStringsProps } from './defs';

const GuitarFretboardStrings = ({ strings }: GuitarFretboardStringsProps) => {
  return (
    <div className="guitar-fretboard-strings">
      {strings.map(({ number }, idx) => (
        <div key={number} className="guitar-fretboard-string-wrapper">
          <div
            className="guitar-fretboard-string"
            style={{ height: idx + 1 + 'px' }}
          />
        </div>
      ))}
    </div>
  );
};

export { GuitarFretboardStrings };
