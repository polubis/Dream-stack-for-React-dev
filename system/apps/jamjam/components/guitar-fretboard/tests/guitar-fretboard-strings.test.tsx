import { render } from '@testing-library/react';

import { GuitarFretboardStrings } from '../guitar-fretboard-strings';

describe('displays guitar fretboard strings when', () => {
  it('[FRAGILE] supported classes and heights are attached', () => {
    const { asFragment } = render(
      <GuitarFretboardStrings
        strings={[
          {
            number: 1,
            notes: [{ id: 1, octave: 2 }],
          },
          {
            number: 2,
            notes: [{ id: 4, octave: 2 }],
          },
          {
            number: 3,
            notes: [{ id: 7, octave: 2 }],
          },
          {
            number: 4,
            notes: [{ id: 11, octave: 2 }],
          },
          {
            number: 5,
            notes: [{ id: 6, octave: 2 }],
          },
          {
            number: 6,
            notes: [{ id: 4, octave: 2 }],
          },
        ]}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
