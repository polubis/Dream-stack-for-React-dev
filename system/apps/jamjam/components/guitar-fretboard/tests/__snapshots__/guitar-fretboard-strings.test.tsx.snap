// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays guitar fretboard strings when [FRAGILE] supported classes and heights are attached 1`] = `
<DocumentFragment>
  <div
    class="guitar-fretboard-strings"
  >
    <div
      class="guitar-fretboard-string-wrapper"
    >
      <div
        class="guitar-fretboard-string"
        style="height: 1px;"
      />
    </div>
    <div
      class="guitar-fretboard-string-wrapper"
    >
      <div
        class="guitar-fretboard-string"
        style="height: 2px;"
      />
    </div>
    <div
      class="guitar-fretboard-string-wrapper"
    >
      <div
        class="guitar-fretboard-string"
        style="height: 3px;"
      />
    </div>
    <div
      class="guitar-fretboard-string-wrapper"
    >
      <div
        class="guitar-fretboard-string"
        style="height: 4px;"
      />
    </div>
    <div
      class="guitar-fretboard-string-wrapper"
    >
      <div
        class="guitar-fretboard-string"
        style="height: 5px;"
      />
    </div>
    <div
      class="guitar-fretboard-string-wrapper"
    >
      <div
        class="guitar-fretboard-string"
        style="height: 6px;"
      />
    </div>
  </div>
</DocumentFragment>
`;
